sound/soc/sof/pm.c

```diff
@@ -128,6 +132,10 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	if (!sof_ops(sdev)->suspend)
 		return 0;
 
+	/* Nothing to do if firmware never booted */
+	if (!sdev->boot_complete)
+		return 0
```

Comment:
well here we may still want to turn the cores off, no?
maybe this should skip to power down instead of returning?

Reply:
@plbossart hmm I missed a big point here. I just realized that we set sdev->boot_complete even if the firmware boot fails. Let me debug this a bit more

Reply:
@plbossart if the FW fails to load upon boot, we call snd_sof_remove() and disable the cores. So I guess we dont need to no anything here.

I've updated the PR to use first_boot instead of boot_complete. first_boot is set to false after the first fw boot is successful.